Index: insns.dat
===================================================================
RCS file: /home/jarusl/nasm-0.98.39-vmx/insns.dat
retrieving revision 1.46
diff -u -p -w -r1.46 insns.dat
--- insns.dat	2 Sep 2003 21:38:48 -0000	1.46
+++ insns.dat	28 Feb 2006 00:43:59 -0000
@@ -1673,3 +1673,17 @@ MOVSHDUP	xmmreg,mem		\301\3\xF3\x0F\x16\
 MOVSHDUP	xmmreg,xmmreg		\3\xF3\x0F\x16\110		PRESCOTT,SSE3
 MOVSLDUP	xmmreg,mem		\301\3\xF3\x0F\x12\110		PRESCOTT,SSE3
 MOVSLDUP	xmmreg,xmmreg		\3\xF3\x0F\x12\110		PRESCOTT,SSE3
+
+; VMX Instructions
+VMCALL		void			\3\x0F\x01\xC1			VMX
+VMCLEAR		mem			\3\x66\x0F\xC7\206		VMX
+VMLAUNCH	void			\3\x0F\x01\xC2			VMX
+VMPTRLD		mem			\2\x0F\xC7\206			VMX
+VMPTRST		mem			\2\x0F\xC7\207			VMX
+VMREAD		mem,reg32		\2\x0F\x78\101			VMX
+VMREAD		reg32,reg32		\2\x0F\x78\101			VMX
+VMRESUME	void			\3\x0F\x01\xC3			VMX
+VMWRITE		reg32,mem		\2\x0F\x79\110			VMX
+VMWRITE		reg32,reg32		\2\x0F\x79\110			VMX
+VMXOFF		void			\3\x0F\x01\xC4			VMX
+VMXON		mem			\3\xF3\x0F\xC7\206		VMX
Index: insns.h
===================================================================
RCS file: /home/jarusl/nasm-0.98.39-vmx/insns.h
retrieving revision 1.31
diff -u -p -w -r1.31 insns.h
--- insns.h	15 Jan 2005 22:15:51 -0000	1.31
+++ insns.h	28 Feb 2006 00:43:59 -0000
@@ -78,6 +78,7 @@ struct itemplate {
 #define IF_SSE    0x00010000UL  /* it's a SSE (KNI, MMX2) instruction */
 #define IF_SSE2   0x00020000UL  /* it's a SSE2 instruction */
 #define IF_SSE3   0x00040000UL  /* it's a SSE3 (PNI) instruction */
+#define IF_VMX	  0x00080000UL  /* it's a VMX instruction */
 #define IF_PMASK  0xFF000000UL  /* the mask for processor types */
 #define IF_PLEVEL 0x0F000000UL  /* the mask for processor instr. level */
                                         /* also the highest possible processor */

